### PR TITLE
Add WIF support for AWS Auth

### PR DESCRIPTION
**Pull request by Zlaticanin**

### CODE REVIEW

**Comment by Zlaticanin** [comment]
Closing this since we need to add support for this in vault-ent: https://github.com/hashicorp/vault-enterprise/pull/5771

File: builtin/credential/aws/path_config_client.go [hunk 1]
```diff
@@ -85,6 +87,12 @@ func (b *backend) pathConfigClient() *framework.Path {
 				Default:     aws.UseServiceDefaultRetries,
 				Description: "Maximum number of retries for recoverable exceptions of AWS APIs",
 			},
+
+			"role_arn": {
+				Type:        framework.TypeString,
+				Default:     "", // TODO: do we need to set the default here?
```
Comment (vinay-gopalan) [inline builtin/credential/aws/path_config_client.go:93]: Looks like the precedent is to set a default for the fields, so I would stick to setting a default for the scope of this PR. I think we can remove the `TODO` here, and can reconsider as needed in a future PR

File: builtin/credential/aws/path_config_client_test.go [hunk 2]
```diff
@@ -129,3 +132,47 @@ func TestBackend_pathConfigClient(t *testing.T) {
 			data["sts_region"], resp.Data["sts_region"])
 	}
 }
+
+// TestBackend_PathConfigRoot_PluginIdentityToken tests that configuration
+// of plugin WIF returns an immediate error.
+func TestBackend_PathConfigRoot_PluginIdentityToken(t *testing.T) {
```
Comment (vinay-gopalan) [inline builtin/credential/aws/path_config_client_test.go:138]:
```suggestion
func TestBackend_pathConfigClient_PluginIdentityToken(t *testing.T) {
```

**Review by vinay-gopalan — APPROVED** [review]
LGTM! Tests pass and code works as expected. Had a couple suggestions

**Pull request merged**